frontend/src/components/Input/index.tsx

```diff
@@ -0,0 +1,31 @@
+import React from 'react';
+
+import { INPUT_TYPE } from '@/constants';
+import { InputType } from '@/types';
+
+import * as S from './index.styles';
+
+export interface InputProps {
+  type?: InputType;
+  isValid?: boolean;
+  hasUnderline?: boolean;
+}
+
+const Input = ({
+  type = INPUT_TYPE.TEXT,
+  isValid = true,
+  hasUnderline = false,
+  ...inputAttribute
+}: InputProps &
+  React.DetailedHTMLProps<React.InputHTMLAttributes<HTMLInputElement>, HTMLInputElement>) => {
+  switch (hasUnderline) {
+    case false:
+      return <S.Input type={type} isValid={isValid} {...inputAttribute} />;
+    case true:
+      return <S.UnderlineInput type={type} isValid={isValid} {...inputAttribute} />;
+    default:
+      return <S.Input type={type} isValid={isValid} {...inputAttribute} />;
+  }
```

Comment:
early return을 사용하면 좀 더 보기 편하고 중복을 줄일 수 있지 않을까요?.?

```suggestion
  if (hasUnderline) {
    return <S.UnderlineInput type={type} isValid={isValid} {...inputAttribute} />;
  }
  
  return <S.Input type={type} isValid={isValid} {...inputAttribute} />;
```

Reply:
나중에 인풋이 추가될때를 대비해서 switch 문으로 작성해줬는데,
굳이 지금 고려할 필요는 없겠네요 :) 수정하겠습니다!